magpie/alembic/versions/2021-01-04_954a9d7fe740_temporary_url_table.py

```diff
@@ -0,0 +1,36 @@
+"""
+Temporary URL table
+
+Revision ID: 954a9d7fe740
+Revises: cfe64807c143
+Create Date: 2021-01-04 12:56:22.298527
+"""
+
+import datetime
+import sqlalchemy as sa
+import uuid
+from alembic import op
+from sqlalchemy.dialects.postgresql import UUID
+
+
+# revision identifiers, used by Alembic.
+revision = "954a9d7fe740"
+down_revision = "cfe64807c143"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.create_table("tmp_tokens",
+                    sa.Column("token", UUID(as_uuid=True), primary_key=True, default=uuid.uuid4, unique=True),
+                    sa.Column("operation", sa.Unicode(32), nullable=False),
+                    sa.Column("user_id", sa.Integer,
+                              sa.ForeignKey("users.id", onupdate="CASCADE", ondelete="CASCADE"), nullable=True),
+                    sa.Column("group_id", sa.Integer(),
```

Comment:
`group_id` and `user_id` are used for the accept T&C `operation`
`user_id` is also used by itself (`group_id = None`) for reset password `operation`

The fields are there is a self-serve fashion. This is why they are allowed to be null.
Whichever information is needed by the `operation` can be defined there, and it up to that `operation` to define expected values properly.
I think the only other "possible" column could be `resource_id`, in case some new `operation` was specific to a resource, but I don't know what that case could be for the moment. 

Since each implementation of `operation` to generate the token are specific to their use case, I think that most of the logic to create the db entry will be in the specific function that needs it.

Reply:
I agree and indeed if only resource_id has to be added in future that's not a huge deal given that we can also add foreign key on it. 
I was thinking to an operation requiring some kind of payload, like click here to apply this random permissions schema. And the schema must be store somehow so that the temporary url could apply it. For that kind of things, I was wondering if we could future proof the db by allowing a blob field where an operation could store whatever it want for its usage?

Reply:
I think it's better to wait for that specific use-case to happen and adjust the db with the most appropriate schema.

Reply:
Correct!